chain/src/store.rs

```diff
@@ -276,7 +276,15 @@ impl Iterator for DifficultyIter {
 			Err(_) => None,
 			Ok(bh) => {
 				self.next = bh.previous;
-				Some(Ok((bh.timestamp.to_timespec().sec as u64, bh.difficulty)))
+
+				// TODO - error handling, base case for the loop, maybe look at height?
+				let difficulty = if let Ok(prev_header) = self.store.get_block_header(&bh.previous) {
```

Comment:
You should be able to use the previous you just loaded in the next iteration?

Reply:
But its the _next_ iteration - we haven't loaded it yet?